release notes beta 5.24.0 (#7192)
hashicorp/terraform-provider-google-beta: CHANGELOG.md

## 5.24.0 (Unreleased)
IMPROVEMENTS:

> * cloudrunv2: added `template.volumes.nfs` field to `google_cloud_run_v2_job` resource (beta) ([#7169](https://github.com/hashicorp/terraform-provider-google-beta/pull/7169))

Reviewer: ```suggestion
* cloudrunv2: added `template.volumes.nfs` field to `google_cloud_run_v2_job` resource ([#7169](https://github.com/hashicorp/terraform-provider-google-beta/pull/7169))
```

* redis: added `node_type` and `precise_size_gb` fields to `google_redis_cluster` ([#7174](https://github.com/hashicorp/terraform-provider-google-beta/pull/7174))
* storage: added `project_number` attribute to `google_storage_bucket` resource and data source ([#7164](https://github.com/hashicorp/terraform-provider-google-beta/pull/7164))
* storage: added ability to provide `project` argument to `google_storage_bucket` data source. This will not impact reading the resource's data, instead this helps users avoid calls to the Compute API within the data source. ([#7164](https://github.com/hashicorp/terraform-provider-google-beta/pull/7164))

BUG FIXES:
* appengine: fixed a crash in `google_app_engine_flexible_app_version` due to the `deployment` field not being returned by the API ([#7175](https://github.com/hashicorp/terraform-provider-google-beta/pull/7175))
* bigquery: fixed a crash when `google_bigquery_table` had a `primary_key.columns` entry set to `""` ([#7166](https://github.com/hashicorp/terraform-provider-google-beta/pull/7166))

> * compute: Fixed update scenarios on `ComputeTargetHttpsProxy` and `ComputeRegionTargetHttpsProxy` resources. ([#7170](https://github.com/hashicorp/terraform-provider-google-beta/pull/7170))

Author: ```suggestion
* compute: fixed update scenarios on `ComputeTargetHttpsProxy` and `ComputeRegionTargetHttpsProxy` resources. ([#7170](https://github.com/hashicorp/terraform-provider-google-beta/pull/7170))
```

> * compute: fixed update scenarios on `ComputeTargetHttpsProxy` and `ComputeRegionTargetHttpsProxy` resources. ([#7170](https://github.com/hashicorp/terraform-provider-google-beta/pull/7170))

Reviewer: Can you change these to the terraform resource names?

Reviewer 2: Resolved

> * compute: Added explicit update_encoder to `ComputeTargetHttpsProxy` and `ComputeRegionTargetHttpsProxy` resources. ([#7170](https://github.com/hashicorp/terraform-provider-google-beta/pull/7170))

Author: ```suggestion
* compute: Fixed update scenarios on `ComputeTargetHttpsProxy` and `ComputeRegionTargetHttpsProxy` resources. ([#7170](https://github.com/hashicorp/terraform-provider-google-beta/pull/7170))
```

> * compute: Added explicit update_encoder to `ComputeTargetHttpsProxy` and `ComputeRegionTargetHttpsProxy` resources. ([#7170](https://github.com/hashicorp/terraform-provider-google-beta/pull/7170))

Reviewer: What does this fix?

> * dataflow: fixed an issue where the provider would crash when `enableStreamingEngine` is set as a `parameter` value in `google_dataflow_flex_template_job` ([#7160](https://github.com/hashicorp/terraform-provider-google-beta/pull/7160))

Author: ```suggestion
* dataflow: fixed an issue where the provider would crash when `enable_streaming_engine` is set as a `parameter` value in `google_dataflow_flex_template_job` ([#7160](https://github.com/hashicorp/terraform-provider-google-beta/pull/7160))
```

> * dataflow: fixed an issue where the provider would crash when `enableStreamingEngine` is set as a `parameter` value in `google_dataflow_flex_template_job` ([#7160](https://github.com/hashicorp/terraform-provider-google-beta/pull/7160))

Reviewer: snake_case the field name
